classes/02class/exercises/c02-network07/jayanath/README.md

```diff
@@ -0,0 +1,216 @@
+      "AssociationState": {
+          "State": "associated"
+      }
+  }
+
+> aws ec2 associate-route-table \
+    --route-table-id rtb-043be3454bd489e40 \
+    --subnet-id subnet-0cf2dc519f5bc9a2a
+
+  {
+      "AssociationId": "rtbassoc-0353f6cc1cb7ae7e0",
+      "AssociationState": {
+          "State": "associated"
+      }
+  }
+
+> aws ec2 associate-route-table \
+    --route-table-id rtb-043be3454bd489e40 \
+    --subnet-id subnet-0462adf3034ba9204
+
+  {
+      "AssociationId": "rtbassoc-02ff139493a5cb206",
+      "AssociationState": {
+          "State": "associated"
+      }
+  }
+
+```
+
+- Commands for creating the following routes:
+
+|route table|destination|target|
+|-|-|-|
+|devopsacademy-rt-public|0.0.0.0|`devopsacademy-igw`|
+|devopsacademy-rt-private|0.0.0.0|`devopsacademy-ngw`|
+
+```
+## add a route to IGW in the public route table
+
+> aws ec2 create-route \
+    --route-table-id rtb-03f1869451bdac82a \
+    --destination-cidr-block 0.0.0.0/0 \
+    --gateway-id igw-0831ac4f89910706b
+
+  {
+      "Return": true
+  }
+
+## add a route to NGW in the private route table
+> aws ec2 create-route \
+    --route-table-id rtb-043be3454bd489e40 \
+    --destination-cidr-block 0.0.0.0/0 \
+    --nat-gateway-id nat-07f98801256171de7
+
+  {
+      "Return": true
+  }
+
+```
+
+- Answer the following questions:
+  - How did you accomplish both instructions?
+    ```
+    Yes. Had to recreate the NAT gateway as I deleted it right after the exercise 06.
+    ```
+
+  - Why did you add 0.0.0.0 route to the IGW on public route table?
+    ```
+    0.0.0.0/0 is the default route. With that, anything that is not locally routable will go out to IGW.
+    ```
+
+  - Why did you add 0.0.0.0 route to the NGW on private route table?
+    ```
+    Same as above. Anything that is not locally routable will go out to NGW as 0.0.0.0/0 is the default route.
+    ```
+
+  - What is the difference of IGW and NGW?
+    ```
+    IGW is a VPC component and a VPC could only have a single IGW. They are highly available and redundant by default.
+    IGW allows internet routable traffic to go out and performs NAT operations for
+    instances with public IPs. Also IGW supports both IPv4 and IPv6 traffic.
+
+    NGW are only scalable in a single AZ, hence multiple NGWs in different AZs are required to maintain redundancy.
+    They allows instances in a private subnet to connect to internet or any other AWS service.
+    However NGW does not allow external entities to initiate connections into those private instances.
+    Also NGW does not support IPv6.
+
+    ```
+
+  - Can you delete the destination route to your VPC network? Why?
+    ```
+    No. We can't delete it. The destination route is there to route local traffic to the VPC router.
+
+    ```
+
+  - What happens if you do not associate your route table with any subnets?
+    ```
+    Nothing. A route table without any associated subnet does not serve any purpose.
```

Comment:
Not quite, if you don't associate a route table with a subnet it will still really on the main rt which by default can't be deleted.